common/browsered/field-validation.js

```diff
@@ -95,23 +98,42 @@ function applyErrorMessaging (form, field, result) {
   let formGroup = field.closest('.form-group')
   if (!formGroup.classList.contains('error')) {
     formGroup.classList.add('error')
-    let label = document.querySelector('label[for="' + field.name + '"]')
-    let errorLabel = label.getAttribute('data-error-label')
-    if (errorLabel) {
-      result = errorLabel
+    const errorLegendElement = formGroup.querySelector('legend')
+    if (errorLegendElement === null) {
+      const errorElement = document.querySelector('label[for="' + field.name + '"]')
+      const errorLabel = errorElement.getAttribute('data-error-label')
+      if (errorLabel) {
+        result = errorLabel
+      }
+      errorElement.insertAdjacentHTML('beforeend',
+        '<span class="error-message">' + result + '</span>')
+    } else {
+      const errorLabel = errorLegendElement.getAttribute('data-error-label')
+      if (errorLabel) {
+        result = errorLabel
+      }
+      const errorElement = document.createElement('span')
+      errorElement.setAttribute('class', 'error-message')
+      errorElement.innerText = result
+      errorLegendElement.appendChild(errorElement)
     }
-    label.insertAdjacentHTML('beforeend',
-      '<span class="error-message">' + result + '</span>')
   }
 }
 
 function populateErrorSummary (form) {
-  let erroringFields = Array.prototype.slice.call(form.querySelectorAll('.form-group.error label'))
-  let configuration = {
-    fields: erroringFields.map(field => {
-      let label = field.innerHTML.split('<')[0].trim()
-      let id = field.getAttribute('for')
-      return {label, id}
+  const erroringformGroups = Array.prototype.slice.call(form.querySelectorAll('.form-group.error'))
+  const configuration = {
+    fields: erroringformGroups.map(formGroup => {
+      const errorLegendElement = formGroup.querySelector('legend')
```

Comment:
Again only using const once so no need to set it

Reply:
Similar question to https://github.com/alphagov/pay-direct-debit-frontend/pull/40#discussion_r165362772, could you please clarify, thanks